### PR TITLE
Gmaps revival

### LINKED ISSUE #3289: use user api key for embeds/public pages with google maps enabled
depends on #3025

### PR DESCRIPTION
**Pull request by javisantana**
Let's get back GMaps!

this is what changed:

- backend:  
    - basemaps are not static anymore, they depend on the user. Basically this allow us to set different basemaps depending if the user has google maps enabled or not. Check user.rb changes https://github.com/CartoDB/cartodb/pull/3337/files#diff-4676c008b11a5480d73d4a6de01e45b9R2352
    - map creation: before app_config.yml configuration was being used, that was changed to be the default map for the user. Check map.rb and table.rb to see the changes. Not needed keys have been removed (so they could be removed from deploy cc @CartoDB/systems )
   
@Kartones I think there are some things to change in the new map.rb

- systems:
   - basemap configuration needs to be changed: https://github.com/CartoDB/cartodb-platform/issues/506

- frontend

  - enabled google maps again, torque, edition and so on is working (@fdans worked on it)
  - google maps baselayer support in the dropdown
  - lots of views changed to include google maps libray when needed

### To make it work:

In order to have google maps you need to:

1. Add the following to the basemaps section in app_config.yml

```
GMaps:
      roadmap:
        name: 'GMaps Roadmap'
        maxZoom: 40
        minZoom: 0
        baseType: 'roadmap'
        className: 'googlemaps'
      hybrid:
        name: 'GMaps Hybrid'
        maxZoom: 40
        minZoom: 0
        baseType: 'hybrid'
        className: 'googlemaps'
      satellite:
        name: 'GMaps Satellite'
        maxZoom: 40
        minZoom: 0
        baseType: 'satellite'
        className: 'googlemaps'
      gray_roadmap:
        name: 'GMaps Gray Roadmap'
        maxZoom: 40
        minZoom: 0
        baseType: 'gray_roadmap'
        className: 'googlemaps'
        style: '[{ "stylers": [ { "saturation": -100 } ] },{ "featureType": "water", "stylers": [ { "gamma": 1.67 }, { "lightness": 27 } ] },{ "elementType": "geometry", "stylers": [ { "gamma": 1.31 }, { "lightness": 12 } ] },{ "featureType": "administrative", "elementType": "labels", "stylers": [ { "lightness": 51 }, { "gamma": 0.94 } ] },{ },{ "featureType": "road", "elementType": "labels", "stylers": [ { "lightness": 57 } ] },{ "featureType": "poi", "elementType": "labels", "stylers": [ { "lightness": 42 } ] }]'
```

2. Enable feature flag ```google_maps```.

---

Fixes #3289.

### CODE REVIEW

**Comment by javisantana** [comment]
@kartones @xavijam @viddo please review

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3373/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3411/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3414/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3416/))

**Comment by javisantana** [comment]
retest this please

File: app/controllers/admin/visualizations_controller.rb [hunk 1]
```diff
@@ -50,6 +50,7 @@ def index
     @tables_count  = current_user.tables.count
     @first_time    = !current_user.dashboard_viewed?
     @just_logged_in = !!flash['logged']
+    @google_maps_api_key = current_user.google_maps_api_key
```
Comment (xavijam) [inline app/controllers/admin/visualizations_controller.rb]: What about if the user belongs to an organization?

**Comment by recena** [comment]
retest this please

**Comment by recena** [comment]
retest this please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3470/))

**Comment by recena** [comment]
@javisantana It seems everything is working fine.

**Pull request merged**